File: ios/USTAQuery/Models/Match.swift
import Foundation

struct SetScore: Codable, Hashable {
    let setNumber: Int
    let playerGames: Int
    let opponentGames: Int
    let tiebreakPlayer: Int?
    let tiebreakOpponent: Int?
}

struct Match: Codable, Identifiable {
    let id: Int
    let tournamentName: String
    let tournamentId: Int
    let divisionName: String
    let round: String
    let matchType: String
    let player1: PlayerSummary
    let player2: PlayerSummary?
    let opponent1Name: String?
    let opponent1Uaid: String?
    let opponent2Name: String?
    let opponent2Uaid: String?
    let winnerSide: String?
    let winType: String?
    let matchDate: String?
    let scoreSummary: String?
    let durationMinutes: Int?
    let sets: [SetScore]
}

File: android/app/src/main/java/com/usta/query/data/model/Match.kt
package com.usta.query.data.model

data class Match(
    val id: Int,
    val tournamentName: String,
    val tournamentId: Int,
    val divisionName: String,
    val round: String,
    val matchType: String,
    val player1: PlayerSummary,
    val player2: PlayerSummary? = null,
    val opponent1Name: String? = null,
    val opponent1Uaid: String? = null,
    val opponent2Name: String? = null,
    val opponent2Uaid: String? = null,
    val winnerSide: String? = null,
    val winType: String? = null,
    val matchDate: String? = null,
    val scoreSummary: String? = null,
    val durationMinutes: Int? = null,
    val sets: List<SetScore> = emptyList()
)

data class SetScore(
    val setNumber: Int,
    val playerGames: Int,
    val opponentGames: Int,
    val tiebreakPlayer: Int? = null,
    val tiebreakOpponent: Int? = null
)
